Gems/Multiplayer/Code/Source/NetworkEntity/NetworkEntityManager.cpp

```diff
@@ -467,6 +467,54 @@ namespace Multiplayer
         return netEntityId;
     }
 
+    AzFramework::EntitySpawnTicket NetworkEntityManager::RequestNetSpawnableInstantiation(const AZ::Data::Asset<AzFramework::Spawnable>& rootSpawnable)
+    {
+        const AzFramework::Spawnable::EntityList& entityList = rootSpawnable->GetEntities();
+        if (entityList.size() == 0)
+        {
+            AZ_Assert(false, "RequestNetSpawnableInstantiation: No entities in the spawnable %s", rootSpawnable.GetHint().c_str())
+            return {};
+        }
+
+        // The first entity in every spawnable is the root one 
+        const AZ::Entity* rootEntity = (entityList.begin())->get();
+
+        const auto* holderComponent = rootEntity->FindComponent<NetworkSpawnableHolderComponent>();
+        if(!holderComponent)
+        {
+            // This spawnable doesn't have a corresponding network spawnable.
+            return {};
+        }
+
+        // Retrieve the corresponding network spawnable asset
+        AZ::Data::Asset<AzFramework::Spawnable> netSpawnableAsset = holderComponent->GetNetworkSpawnableAsset();
+
+        // Prepare the parameters for the spawning process
+        AzFramework::SpawnAllEntitiesOptionalArgs optionalArgs;
+        optionalArgs.m_priority = AzFramework::SpawnablePriority_High;
+
+        // Pre-insertion callback allows us to do network-specific setup for the entities before they are added to the scene
+        optionalArgs.m_preInsertionCallback = [spawnableAssetId = netSpawnableAsset.GetId()](AzFramework::EntitySpawnTicket::Id,
+            AzFramework::SpawnableEntityContainerView entities)
+        {
+            for (uint32_t netEntityIndex = 0; netEntityIndex < entities.size(); netEntityIndex++)
+            {
+                AZ::Entity* netEntity = *(entities.begin() + netEntityIndex);
```

Comment:
Doesn't `SpawnableEntityContainerView` provide a proper function to access elements of the container?
If it doesn't, maybe it could be added.

Reply:
It doesn't and looks like this method of working with iterators is widely used everywhere else with SpawnableEntityContainerView. 
Possibly it is intentional.

Reply:
The common use case was to use a ranged for-loop, so it simply wasn't added because an index operator wasn't needed. Feel free to add it to `SpawnableEntityContainerView` and `SpawnableConstEntityContainerView` or add a ticket so it'll get added in the future.

Reply:
https://github.com/o3de/o3de/issues/4110